### PR TITLE
Fix memory format

### LINKED ISSUE #53: `get_memory()` fails because memory has wrong format
With qiskit providers, the `memory` attribute of a task result is a list of strings with the measurement outputs. With the braket provider however it is a list of numpy arrays of integers. This causes `task.result().get_memory()` to fail.

### Steps to reproduce the problem
* Run a task with `memory=True`
* Execute `task.result().get_memory()`

### What is the current behavior?

```
AttributeError: 'numpy.ndarray' object has no attribute 'startswith'
```

### What is the expected behavior?

Return the memory

### LINKED ISSUE #59: `memory = True` argument in function `execute` is not working.
### The problem
The argument `memory = True` in the `execute` function is not working, hence per-shot measurement result is not accessible.

### Steps to reproduce the problem
1. Create a circuit in qiskit and define a backend from`aws_braket_provider`.
2. Run the job using `execute` function with `memory =  True`.
3. Extract the memory using `result.get_memory()` method.

Example: 
```
qc = QuantumCircuit(2)
qc.h(0)
qc.cx(0,1)
qc.measure_all()

ionq = aws_provider.get_backend('IonQ Device')
tqc = transpile(qc, backend = ionq)

job = execute(tqc, backend = ionq, shots = 1024, memory = True)
result = job.result()
counts = result.get_counts()
memory = result.get_memory()
```
### What is the current behavior?
Try to extract memory gives the following error:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Input In [19], in <cell line: 3>()
      1 result = job.result()
      2 counts = result.get_counts()
----> 3 memory = result.get_memory()

File ~/anaconda3/envs/qlab/lib/python3.10/site-packages/qiskit/result/result.py:229, in Result.get_memory(self, experiment)
    226 memory = self.data(experiment)["memory"]
    228 if meas_level == MeasLevel.CLASSIFIED:
--> 229     return postprocess.format_level_2_memory(memory, header)
    230 elif meas_level == MeasLevel.KERNELED:
    231     return postprocess.format_level_1_memory(memory)

File ~/anaconda3/envs/qlab/lib/python3.10/site-packages/qiskit/result/postprocess.py:151, in format_level_2_memory(memory, header)
    149 memory_list = []
    150 for shot_memory in memory:
--> 151     memory_list.append(format_counts_memory(shot_memory, header))
    152 return memory_list

File ~/anaconda3/envs/qlab/lib/python3.10/site-packages/qiskit/result/postprocess.py:65, in format_counts_memory(shot_memory, header)
     46 def format_counts_memory(shot_memory, header=None):
     47     """
     48     Format a single bitstring (memory) from a single shot experiment.
     49 
   (...)
     63         dict: a formatted memory
     64     """
---> 65     if shot_memory.startswith("0x"):
     66         shot_memory = _hex_to_bin(shot_memory)
     67     if header:

AttributeError: 'numpy.ndarray' object has no attribute 'startswith'
```
### What is the expected behavior?
The `result.get_memory()` should return a list of measurement result of each shot.

### A work around
This [answer](https://quantumcomputing.stackexchange.com/a/27854/19631) on stackexchange suggest a way around which works well for simulators from `AWSBraketProvider`, but for `IonQ Device` it is not useful.

### LINKED ISSUE #67: TestAWSBraketJob using incorrect assert calls
These five `assertTrue` calls appear to be likely typos, and should probably be `assertEqual` in order to verify the intended behavior.

https://github.com/qiskit-community/qiskit-braket-provider/blob/825f8eca0310279dab3c02bfd06894f165ed7e1c/tests/providers/test_braket_job.py#L24-L29

### PR DESCRIPTION
**Pull request by jcjaskula-aws**
### Summary

The Braket `result().measurement` field returns shot results using np.array(). Qiskit expect memory to be a list of bitstring. This MR modifies how the results are passed to ExperimentResultData.

This PR also fixes test_braket_job unittest.

### Details and comments
Closes #53 
Closes #59 (as memory=True is not needed)

Closes #67

### CODE REVIEW

File: tests/providers/test_braket_job.py [hunk 1]
```diff
@@ -25,5 +25,6 @@ def test_job(self):
         self.assertTrue(job.result().results[0].data.counts, {"00": 1})
         self.assertTrue(job.result().results[0].shots, 100)
         self.assertTrue(job.result().results[0].status, "COMPLETED")
+        self.assertTrue(job.result().get_memory(), ["00"])
```
Comment (IceKhan13) [inline tests/providers/test_braket_job.py:28]: I think it should be `assertEqual` as was pointed out here https://github.com/qiskit-community/qiskit-braket-provider/issues/67. 

If you can change it your PR will close 3 issues :) Amazing job!

**Review by IceKhan13 — CHANGES_REQUESTED** [review]
Thank you for PR @jcjaskula-aws !

Amazing job!

Only one tiny change, which will help close another issue!

File: tests/providers/test_braket_job.py (hunk 1 above)
Comment (jcjaskula-aws) [reply tests/providers/test_braket_job.py:28]: Sounds good. I was expected #67 to be fixed in another PR but I can do this later.

**Review by IceKhan13 — APPROVED** [review]
3-pointer PR :) 

3pointer

**Pull request merged**